src/Presentation/DonationConfirmationPageSelector.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace WMDE\Fundraising\Frontend\Presentation;
+
+use WMDE\Fundraising\Frontend\Domain\SelectedConfirmationPage;
+use WMDE\Fundraising\Frontend\Domain\TemplateTestCampaign;
+
+/**
+ * Selects a random confirmation page from given options
+ *
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class DonationConfirmationPageSelector {
+
+	private $defaultPageTitle;
+	/** @var TemplateTestCampaign[] */
+	private $campaigns;
+
+	public function __construct( array $config ) {
```

Comment:
Gave this a try, not convinced this is actually better here given $campaigns still has the config format